src/SolverInterface/attributes.jl

```diff
@@ -0,0 +1,250 @@
+
+Set an attribute of the model `m` specified by attribute type `attr`.
+"""
+function setattribute! end
+
+# Scalar attributes
+"""
+    ObjectiveValue
+
+The objective value of the best primal result vector found by the solver.
+"""
+struct ObjectiveValue <: AbstractAttribute end
+
+"""
+    ObjectiveBound
+
+The best known bound on the optimal objective value.
+"""
+struct ObjectiveBound <: AbstractAttribute end
+
+
+struct RelativeGapAttr <: AbstractAttribute  end
+"""
+    RelativeGap
+
+The final relative optimality gap as optimization terminated. That is, ``\\frac{|b-f|}{|f|}``, where ``b`` is the best bound and ``f`` is the best feasible objective value.
+"""
+const RelativeGap = RelativeGapAttr()
+
+"""
+    SolveTime
+
+The total elapsed solution time (in seconds) as reported by the solver.
+"""
+struct SolveTime <: AbstractAttribute end
+
+"""
+    Sense
+
+The optimization sense of the model, an `OptimizationSense` with value `MinSense` or `MaxSense`.
+"""
+struct Sense <: AbstractAttribute end
+
+@enum OptimizationSense MinSense MaxSense
+
+"""
+    SimplexIterations
+
+The cumulative number of simplex iterations during the optimization process. In particular, for a MIP the total simplex iterations for all nodes.
+"""
+struct SimplexIterations <: AbstractAttribute end
+
+"""
+    BarrierIterations
+
+The cumulative number of barrier iterations during the optimization process.
+"""
+struct BarrierIterations <: AbstractAttribute end
+
+"""
+    NodeCount
+
+The total number of branch-and-bound nodes explored.
+"""
+struct NodeCount <: AbstractAttribute end
+
+"""
+    RawSolver
+
+An object that may be used to access a solver-specific API for this model.
+"""
+struct RawSolver <: AbstractAttribute end
+
+"""
+    ResultCount
+
+The number of results available.
+"""
+struct ResultCount <: AbstractAttribute end
+
+
+# Variable attributes
+
+"""
+    VariableStart
+
+An initial assignment of the variables that the solver may use
+to warm-start the solve.
+"""
+struct VariableStart <: AbstractAttribute end
+
+"""
+    VariableLowerBoundDualStart
+
+An initial assignment of the variable lower-bound duals that the solver may use
+to warm-start the solve.
+"""
+struct VariableLowerBoundDualStart <: AbstractAttribute end
+
+"""
+    VariableUpperBoundDualStart
+
+An initial assignment of the variable upper-bound duals that the solver may use
+to warm-start the solve.
+"""
+struct VariableUpperBoundDualStart <: AbstractAttribute end
+
+
+"""
+    VariableLowerBound
+
+Lower-bound constraints on variables. `-Inf` is valid as no bound.
+"""
+struct VariableLowerBound <: AbstractAttribute end
+
+
+"""
+    VariableUpperBound
+
+Upper-bound constraints for the variables. `Inf` is valid as no bound.
+"""
+struct VariableUpperBound <: AbstractAttribute end
+
+"""
+    VariablePrimal(N)
+
+The assignment to the primal variables in result `N`. If `N` is omitted, it is interpreted as 1.
```

Comment:
Is the plan to implement this with a ``VariablePrimal() = VariablePrimal(1)`` constructor?